hash.py

```diff
@@ -0,0 +1,34 @@
+class Hash(object):
+    def __init__(self, buckets):
+        self.h_list = []
+        for i in range(0, buckets):
```

Comment:
If the start value is 0 you don't need to specify it.  That makes this whole expression a bit cleaner if you say something like:

```python
self.h_list = [[] for _ in range(buckets)]
```